### PR TITLE
Wait for the check to signal he ran before checking his status

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

Wait for the check to signal he ran before checking his status

### CODE REVIEW

File: pkg/collector/runner/runner_test.go [hunk 1]
```diff
@@ -33,6 +42,7 @@ func (c *TestCheck) Interval() time.Duration                            { return
 func (c *TestCheck) Run() error {
 	c.Lock()
 	defer c.Unlock()
+	defer func() { c.done <- nil }()
```
Comment (remicalixte) [inline pkg/collector/runner/runner_test.go]: You can just close the channel here I think

File: pkg/collector/runner/runner_test.go [hunk 2]
```diff
@@ -23,6 +24,14 @@ type TestCheck struct {
 	sync.Mutex
```
Comment (remicalixte) [inline pkg/collector/runner/runner_test.go]: Is the mutex still useful with the channel?

Comment (hush-hush) [reply pkg/collector/runner/runner_test.go]: We still use `hasRun` in other tests (ie num_worker_test) so I think so

File: pkg/collector/runner/runner_test.go (hunk 1 above)
Comment (hush-hush) [reply pkg/collector/runner/runner_test.go]: :+1:

File: pkg/collector/runner/runner_test.go (hunk 2 above)
Comment (remicalixte) [reply pkg/collector/runner/runner_test.go]: Yes right

**Comment by remicalixte** [comment]
It seems that there is an other issue here: `Aggregator was not initialized`

**Comment by remicalixte** [comment]
I can't reproduce it locally...

**Comment by hush-hush** [comment]
@remicalixte The checks are sending metrics to the aggregator, which is not initialized. It's expected.

**Comment by remicalixte** [comment]
Oh okay, but it's not expected to spend more than 1min

**Review by remicalixte — APPROVED** [review]

**Pull request merged**